sentry/src/main/java/io/sentry/metrics/MetricResourceIdentifier.java

```diff
@@ -0,0 +1,64 @@
+package io.sentry.metrics;
+
+import io.sentry.MeasurementUnit;
+import io.sentry.util.Objects;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Uniquely identifies a metric resource. Used for caching the {@link CodeLocations} for a given
+ * metric.
+ */
+@ApiStatus.Internal
+public final class MetricResourceIdentifier {
```

Comment:
just a small note: `MetricResourceIdentifier` and `CodeLocations` are not referenced anywhere.
I understand that they are going to be used in the next steps, so it's fine to keep them, but it would make the PR cleaner to avoid them now